Logging systests:  'list_metrics' returning extra entries

See the [first error here](https://circleci.com/gh/GoogleCloudPlatform/google-cloud-python/6995):

```python
________________________ TestLogging.test_list_metrics _________________________

self = <test_system.TestLogging testMethod=test_list_metrics>

    def test_list_metrics(self):
        METRIC_NAME = 'test-list-metrics%s' % (_RESOURCE_ID,)
        metric = Config.CLIENT.metric(
            METRIC_NAME, DEFAULT_FILTER, DEFAULT_DESCRIPTION)
        self.assertFalse(metric.exists())
        before_metrics = list(Config.CLIENT.list_metrics())
        before_names = set(metric.name for metric in before_metrics)
        metric.create()
        self.to_delete.append(metric)
        self.assertTrue(metric.exists())
        after_metrics = list(Config.CLIENT.list_metrics())
        after_names = set(metric.name for metric in after_metrics)
        self.assertEqual(after_names - before_names,
>                        set([METRIC_NAME]))
E       AssertionError: Items in the first set but not the second:
E       u'test-create-metric-6992-1529965066'

tests/system/test_system.py:343: AssertionError
```

My hypothesis is that overlapping CI runs are causing the extra metrics instances to be present.

--- a/logging/tests/system/test_system.py
+++ b/logging/tests/system/test_system.py
@@ -334,14 +334,14 @@ class TestLogging(unittest.TestCase):
             METRIC_NAME, DEFAULT_FILTER, DEFAULT_DESCRIPTION)
         self.assertFalse(metric.exists())
         before_metrics = list(Config.CLIENT.list_metrics())
-        before_names = set(metric.name for metric in before_metrics)
+        before_names = set(before.name for before in before_metrics)
+        self.failIf(metric.name in before_names)
         metric.create()
         self.to_delete.append(metric)
         self.assertTrue(metric.exists())
         after_metrics = list(Config.CLIENT.list_metrics())
-        after_names = set(metric.name for metric in after_metrics)
-        self.assertEqual(after_names - before_names,
-                         set([METRIC_NAME]))
+        after_names = set(after.name for after in after_metrics)
+        self.assertTrue(metric.name in after_names)
 
     def test_reload_metric(self):
         METRIC_NAME = 'test-reload-metric%s' % (_RESOURCE_ID,)

--- a/logging/tests/system/test_system.py
+++ b/logging/tests/system/test_system.py
@@ -472,14 +472,14 @@ class TestLogging(unittest.TestCase):
         sink = Config.CLIENT.sink(SINK_NAME, DEFAULT_FILTER, uri)
         self.assertFalse(sink.exists())
         before_sinks = list(Config.CLIENT.list_sinks())
-        before_names = set(sink.name for sink in before_sinks)
+        before_names = set(before.name for before in before_sinks)
+        self.failIf(sink.name in before_names)
         sink.create()
         self.to_delete.append(sink)
         self.assertTrue(sink.exists())
         after_sinks = list(Config.CLIENT.list_sinks())
-        after_names = set(sink.name for sink in after_sinks)
-        self.assertEqual(after_names - before_names,
-                         set([SINK_NAME]))
+        after_names = set(after.name for after in after_sinks)
+        self.assertTrue(sink.name in after_names)
 
     def test_reload_sink(self):
         SINK_NAME = 'test-reload-sink%s' % (_RESOURCE_ID,)

--- a/logging/tests/system/test_system.py
+++ b/logging/tests/system/test_system.py
@@ -231,9 +231,10 @@ class TestLogging(unittest.TestCase):
     def test_log_handler_async(self):
         LOG_MESSAGE = 'It was the worst of times'
 
-        handler = CloudLoggingHandler(Config.CLIENT)
+        handler_name = 'gcp-async' + unique_resource_id('-')
+        handler = CloudLoggingHandler(Config.CLIENT, name=handler_name)
         # only create the logger to delete, hidden otherwise
-        logger = Config.CLIENT.logger(handler.name)
+        logger = Config.CLIENT.logger(handler_name)
         self.to_delete.append(logger)
 
         cloud_logger = logging.getLogger(handler.name)

